packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -0,0 +1,197 @@
+import { Construct } from 'constructs';
+import { CfnFleet } from './codebuild.generated';
+import { ComputeType } from './compute-type';
+import { EnvironmentType } from './environment-type';
+import { Arn, ArnFormat, IResource, Resource, Token } from '../../core';
+
+/**
+ * Construction properties of a CodeBuild {@link Fleet}.
+ */
+export interface FleetProps {
+  /**
+   * The name of the Fleet.
+   *
+   * @default - CloudFormation generated name
+   */
+  readonly fleetName?: string;
+
+  /**
+   * The number of machines allocated to the compute ﬂeet.
+   * Deﬁnes the number of builds that can run in parallel.
+   *
+   * Minimum value of 1.
+   */
+  readonly baseCapacity: number;
+
+  /**
+   * The instance type of the compute fleet.
+   */
+  readonly computeType: FleetComputeType;
+
+  /**
+   * The environment type of the fleet.
+   */
+  readonly environmentType: EnvironmentType;
+}
+
+/**
+ * Represents a {@link Fleet} for a reserved capacity CodeBuild project.
+ */
+export interface IFleet extends IResource {
+  /**
+   * The ARN of the fleet.
+   * @attribute
+   */
+  readonly fleetArn: string;
+
+  /**
+   * The name of the fleet.
+   * @attribute
+   */
+  readonly fleetName: string;
+
+  /**
+   * The compute type of the fleet.
+   */
```

Comment:
Can we make this more descriptive? The compute type is describing the maximum number of vCPUs available, so maybe

```
The maximum size of the fleet
```

Or something better?

Reply:
The vCPU/memory also depend on the `environmentType`, so it's not really fully describing the resource allocation. I also feel like "maximum size" sounds too much like capacity, as in the number of instances spawned in the fleet.

I tried looking at how `computeType` was described in the project environment, it's not a lot more helpful:

https://github.com/aws/aws-cdk/blob/75038997c96c2da6c5a8ffd5f5eb3f50db4f09c4/packages/aws-cdk-lib/aws-codebuild/lib/project.ts#L1623-L1629

Reply:
can we link to the enum here? While it's not very helpful, the only way to really explain what this property type is is by reading the enum values

Reply:
Done